.github/workflows/ci.yml

```diff
@@ -72,21 +72,40 @@ jobs:
         env:
           RUSTFLAGS: "-Awarnings"       # Suppress warnings
           DATABASE_URL: "postgresql://admin:admin@localhost:5432/defaultdb"  # Set database URL for tests
-        run: |       
-          echo "install sqlx... "
+          GETH_URL: "ws://localhost:8546" # Set local setup Geth Endpoint 
+        run: |
+          echo "Install dependencies for Geth on local"
```

Comment:
As the Ethereum analysis project grows, its codebase, test cases, and CI/CD pipeline execution logic are becoming increasingly complex. To mitigate this complexity, it would be beneficial to extract the project into an independent repository. This separation will help prevent new commits from causing test failures that block the entire CI/CD pipeline.

Furthermore, CI/CD commands should be executed in isolated jobs rather than combining them into a single sequence of shell commands within a ‘run’ step. This approach improves job parallelism and maintainability. It’s also important to note that tutorials 1, 2, 3, 4, and 5 are largely independent of each other, so they can be executed in parallel without interdependencies.